julia/NIfTIZarr.jl/src/header.jl

```diff
@@ -181,209 +93,237 @@ function _read!(io::IOType, header::NiftiHeader)
+)
+
+
+IntentNbPrm = Dict(key => 0 for (key, v) in IntentCode)
 merge!(IntentNbPrm, Dict(
-    IntentCode.CORREL => 1,
-    IntentCode.TTEST => 1,
-    IntentCode.FTEST => 2,
-    IntentCode.CHISQ => 1,
-    IntentCode.BETA => 2,
-    IntentCode.BINOM => 2,
-    IntentCode.GAMMA => 2,
-    IntentCode.POISSON => 1,
-    IntentCode.NORMAL => 2,
-    IntentCode.FTEST_NONC => 3,
-    IntentCode.CHISQ_NONC => 2,
-    IntentCode.LOGISTIC => 2,
-    IntentCode.LAPLACE => 2,
-    IntentCode.UNIFORM => 2,
-    IntentCode.TTEST_NONC => 2,
-    IntentCode.WEIBULL => 2,
-    IntentCode.CHI => 1,
-    IntentCode.INVGAUSS => 2,
-    IntentCode.EXTVAL => 2,
-    IntentCode.GENMATRIX => 2,
-    IntentCode.SYMMATRIX => 1,
+    "corr" => 1,
+    "ttest" => 1,
+    "ftest" => 2,
+    "chi2" => 1,
+    "beta" => 2,
+    "binomial" => 2,
+    "gamma" => 2,
+    "poisson" => 1,
+    "normal" => 2,
+    "ncftest" => 3,
+    "ncchi2" => 2,
+    "logistic" => 2,
+    "laplace" => 2,
+    "uniform" => 2,
+    "ncttest" => 2,
+    "weibull" => 2,
+    "chi" => 1,
+    "invgauss" => 2,
+    "extval" => 2,
+    "matrix" => 2,
+    "symmatrix" => 1,
 ))
 
-@enumx XFormCode begin
-    UNKNOWN = 0
-    SCANNER_ANAT = 1
-    ALIGNED_ANAT = 2
-    TALAIRACH = 3
-    MNI = 4
-    TEMPLATE_OTHER = 5
-end
-
-@enumx SliceOrderCode begin
-    UNKNOWN = 0
-    SEQ_INC = 1
-    SEQ_DEC = 2
-    ALT_INC = 3
-    ALT_DEC = 4
-    ALT_INC2 = 5
-    ALT_DEC2 = 6
-end
-
-UnitRecoder(x::Integer) = string(UnitCode.T(x))
-UnitRecoder(x::String) = getproperty(UnitCode, Symbol(x))
-IntentRecoder(x::Integer) = string(IntentCode.T(x))
-IntentRecoder(x::String) = getproperty(IntentCode, Symbol(x))
-XFormRecoder(x::Integer) = string(XFormCode.T(x))
-XFormRecoder(x::String) = getproperty(XFormCode, Symbol(x))
-SliceOrderRecoder(x::Integer) = string(SliceOrderCode.T(x))
-SliceOrderRecoder(x::String) = getproperty(SliceOrderCode, Symbol(x))
+XFormCode = Dict(
+    "" => 0,
+    "scanner_anat" => 1,
+    "aligned_anat" => 2,
+    "talairach" => 3,
+    "mni_152" => 4,
+    "template_other" => 5
+)
+
+SliceOrderCode = Dict(
+    "" => 0,
+    "seq+" => 1,
+    "seq-" => 2,
+    "alt+" => 3,
+    "alt-" => 4,
+    "alt2+" => 5,
+    "alt2-" => 6
+)
+
+
+DataTypeCode = Bijection(DataTypeCode)
+UnitCode = Bijection(UnitCode)
+IntentCode = Bijection(IntentCode)
+XFormCode = Bijection(XFormCode)
+SliceOrderCode = Bijection(SliceOrderCode)
+
+UnitRecoder(x::Integer) = UnitCode(x)
+UnitRecoder(x::String) = UnitCode[x]
+IntentRecoder(x::Integer) = IntentCode(x)
+IntentRecoder(x::String) = IntentCode[x]
+XFormRecoder(x::Integer) = XFormCode(x)
+XFormRecoder(x::String) = XFormCode[x]
+SliceOrderRecoder(x::Integer) = SliceOrderCode(x)
+SliceOrderRecoder(x::String) = SliceOrderCode[x]
 
 function DataTypeRecoder(x::Integer)
-    if x == DataTypeCode.rgb
-        (("r", "u1"), ("g", "u1"), ("b", "u1"))
-    elseif x == DataTypeCode.rgba
-        (("r", "u1"), ("g", "u1"), ("b", "u1"), ("a", "u1"))
```

Comment:
I thought this was taken from the ome-zarr spec. Was I wrong or is it one of these things that has changed recently?